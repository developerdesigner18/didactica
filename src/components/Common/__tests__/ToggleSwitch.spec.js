import { createLocalVue, mount } from "@vue/test-utils";
import { waitRAF } from "@/../tests/utils.js";

import LxpToggleSwitch from "../ToggleSwitch.vue";

// create an extended `Vue` constructor
const localVue = createLocalVue();

localVue.component("lxp-toggle-switch", LxpToggleSwitch);

describe("ToggleSwitch.vue", () => {
  it("render when created ", async () => {
    let wrapper = await getWrapper();
    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper.element).toMatchSnapshot();
    wrapper.destroy();
  });

  describe("Props", () => {
    let wrapper;

    afterEach(() => {
      wrapper.destroy();
    });
    it("should update based on `toggleState`", async () => {
      wrapper = await getWrapper();
      await wrapper.setProps({
        toggleState: true
      });
      await waitRAF();
      const toggleCheckbox = wrapper.find("#switch");
      expect(toggleCheckbox.element.value).toBeTruthy();
    });
    it("should update based on `disabled` prop", async () => {
      wrapper = await getWrapper();
      await wrapper.setProps({
        disabled: true
      });
      await waitRAF();
      const imgPath = wrapper.find(".lxp-toggle__disabled");
      expect(imgPath.exists()).toBe(true);
    });
  });
});

function getWrapper() {
  let wrapper = mount(LxpToggleSwitch, {
    propsData: {
      toggleState: false
    },
    localVue
  });

  return wrapper;
}
