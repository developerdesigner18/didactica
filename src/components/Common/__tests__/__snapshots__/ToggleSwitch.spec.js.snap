// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch.vue render when created  1`] = `
<div
  class="lxp-toggle"
>
  <input
    id="switch"
    type="checkbox"
  />
   
  <label
    for="switch"
  >
    Toggle
  </label>
</div>
`;
